controller/array_action/array_mediator_ds8k.py

```diff
@@ -279,27 +283,104 @@ def get_volume(self, name, volume_context=None, volume_prefix=""):
+        volume_id = get_volume_id_from_scsi_identifier(volume_id)
+        try:
+            host_name_to_lun_id = {}
+            for host in self.client.get_hosts():
+                host_mappings = host.mappings_briefs
+                for mapping in host_mappings:
+                    if volume_id == mapping["volume_id"]:
+                        host_name_to_lun_id[host.name] = mapping["lunid"]
+                        break
+            logger.debug("Found volume mappings: {}".format(host_name_to_lun_id))
+            return host_name_to_lun_id
+        except exceptions.ClientException as ex:
+            logger.error(
+                "Failed to get volume mappings. Reason is: {}".format(ex.details)
+            )
+            raise ex
 
     def map_volume(self, volume_id, host_name):
-        # TODO: CSI-1198
-        pass
+        logger.debug("Mapping volume {} to host {}".format(volume_id, host_name))
+        scsi_id = volume_id
+        volume_id = get_volume_id_from_scsi_identifier(volume_id)
+        try:
+            mapping = self.client.map_volume_to_host(host_name, volume_id)
+            lun = scsilun_to_int(mapping.lunid)
+            logger.debug("Successfully mapped volume to host with lun {}".format(lun))
+            return lun
+        except exceptions.NotFound:
+            raise array_errors.HostNotFoundError(host_name)
+        except exceptions.ClientException as ex:
+            # [BE586015] addLunMappings Volume group operation failure: volume does not exist.
+            if ERROR_CODE_VOLUME_NOT_FOUND_FOR_MAPPING in str(ex.message).upper():
+                raise array_errors.VolumeNotFoundError(scsi_id)
+            else:
+                raise array_errors.MappingError(scsi_id, host_name, ex.details)
 
     def unmap_volume(self, volume_id, host_name):
-        # TODO: CSI-1199
-        pass
+        logger.debug("Unmapping volume {} from host {}".format(volume_id, host_name))
+        scsi_id = volume_id
+        volume_id = get_volume_id_from_scsi_identifier(volume_id)
+        try:
+            mappings = self.client.get_host_mappings(host_name)
+            lunid = None
+            for mapping in mappings:
+                if mapping.volume == volume_id:
+                    lunid = mapping.lunid
+                    break
+            if lunid is not None:
+                self.client.unmap_volume_from_host(
+                    host_name=host_name,
+                    lunid=lunid
+                )
+                logger.debug("Successfully unmapped volume from host.")
+            else:
+                raise array_errors.VolumeNotFoundError(scsi_id)
+        except exceptions.NotFound:
+            raise array_errors.HostNotFoundError(host_name)
+        except exceptions.ClientException as ex:
+            raise array_errors.UnMappingError(scsi_id, host_name, ex.details)
 
     def get_array_iqns(self):
         return []
 
     def get_array_fc_wwns(self, host_name=None):
```

Comment:
doesn't match? please check again.

Reply:
I checked again:
![image](https://user-images.githubusercontent.com/57443811/76543575-64a72d00-648f-11ea-900c-02cb8f8a9a0c.png)

Reply:
Perhaps it would be more clear if my comment was in the interface:
![image](https://user-images.githubusercontent.com/57443811/76543823-b9e33e80-648f-11ea-9eff-77a6f4b40915.png)

Reply:
it is not the scope of this pr, this pr is for ds8k actions.

Reply:
oh, right, it is related to xiv change.